docs/DeveloperGuide.md

```diff
@@ -1,34 +1,938 @@
+    1. Click `Configure` → `Project Defaults` → `Project Structure`.
+    2. Click `New…` and set it to the directory of the JDK.
+4. Click `Import Project` (or `Open or Import` in newer version of Intellij).
+5. Locate the `build.gradle` file (not the root folder as you would do in a normal importing) and select it. Click `OK`.
+If asked, choose to `Open as Project` (not `Open as File`).
+7. Click `OK` to accept the default settings
+
+--------------------------------------------------------------------------------------------------------------------
+
+## 3. Design (Architecture)
+(Zeyu)
+
+![Architecture Diagram of Design Component](UML/Architecture.png)
+
+The Architecture Diagram given above explains the high-level design of the App. Given below is a quick overview of each component.
+
+The main class is `Kaji.java`. It is responsible for:
+
+* Initializing the components in the correct sequence, and connects them up with each other at app launch.
+* Shutting down the components and invoking cleanup methods where necessary at exit.
+
+The rest of the App consists of 8 components:
+* `Access`: Keeps track of the access level of the user.
+* `UI`: The UI of the App.
+* `Command`: Executes the different command types.
+* `Common`: Contains common classes.
+* `Manager`: Holds the data of the App in memory.
+* `Paser`: Parses user input into specific command type.
+* `Scheduler`: Schedules the revision schedule.
+* `Storage`: Reads data from, and writes data to, the hard disk.
+
+### 3.1. Ui Component 
+(Jia Ern)
+
+![Class Diagram of Ui Component](DG_Images/ui_component.PNG)
+
+The UI component consists of a main class — Ui.java. The Kaji, Logic and Storage components have a dependency on the UI component due to the need to take in user input and show the results of execution.
+
+The UI component is responsible for:
+* Taking in user input.
+* Printing result of execution for the different commands. 
+* Printing error messages when certain exceptions occur. 
+
+### 3.2. Logic Component 
+(Jane)
+
+The Logic component consists of the `Parser`, `Command` and `Scheduler` classes.
+
+<p align="center">
+  <img src="DG_Images/LogicClassDiagram.png" width="600" alt="Logic Class Diagram"/>
+  <br/>Figure <>. Class diagram of Logic component  
+</p>
+
+1. `Kaji` uses the `Parser` class to parse the user command.
+2. This results in a `Command` object which is executed by `Kaji`.
+3. The command execution can affect the `Model` and `Storage` (e.g. adding a module).
+4. The `Scheduler` is used in some command execution to schedule the due date of a chapter.
+5. The result of the command execution is passed to the `Ui`.
+
+Given below is the Sequence Diagram for interactions within the `Logic` component for the `parse("edit 1 CS2113T")` API call:
+
+<p align="center">
+  <img src="DG_Images/LogicSequenceDiagram.png" width="600" alt="Logic Sequence Diagram"/>
+  <br/>Figure <>. Sequence diagram of Logic component  
+</p>
```

Comment:
Maybe you should standardize the way you add diagrams? Right now some of them are aligned and properly labelled with captions while others are not.